src/node/in_memory_ext.rs

```diff
@@ -742,17 +743,22 @@ mod tests {
         let expected_response = new_timestamp;
 
         let actual_response = node
-            .set_next_block_timestamp(new_timestamp)
+            .set_next_block_timestamp(new_timestamp.into())
             .expect("failed setting timestamp");
         let timestamp_after = node
             .get_inner()
             .read()
             .map(|inner| inner.current_timestamp)
             .expect("failed reading timestamp");
 
-        assert_eq!(expected_response, actual_response, "erroneous response");
         assert_eq!(
-            new_timestamp, timestamp_after,
+            expected_response,
+            actual_response.as_u64(),
+            "erroneous response"
+        );
+        assert_eq!(
+            new_timestamp,
+            timestamp_after + 1,
```

Comment:
why `+1`?

Reply:
See above - node's current timestamp is 1 less than the timestamp of the next block.